src/js/game/LevelMVC/LevelView.js

```diff
@@ -1177,6 +1178,13 @@ module.exports = class LevelView {
     positions.forEach(position => {
       if (position) {
         const newBlock = this.controller.levelModel.actionPlane.getBlockAt(position);
+
+        //we don't want to refresh doors. They're not destroyable, and refreshing
+        //will lead to bad animation states
+        if (newBlock.blockType.startsWith("door")) {
```

Comment:
I'd love to have a less fragile way of determining whether or not something is a door, even if that just means putting this exact same check into a `newBlock.isDoor()` method

Reply:
That's fair. Checking strings is not my favorite method of verifying something. However, this whole code base has, what I'd comfortably guess as -hundreds-, of blockType checks to dictate various behavior. I can easily extrapolate that out, just a heads up though.

Josh and I have spoken quite a bit about how much this code base could benefit from some heavy house-keeping. Just probably best to tackle the systemic stuff while we're not working on an active release. 😄

Reply:
Yeah, I've been seeing a lot of it. I am very much in favor of the idea of trying to make small housekeeping improvements as we go, since it's so much harder to schedule that kind of work after the product is finished.

That being said, consider this comment just a suggestion.